org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/InitializationContext.java

```diff
@@ -21,4 +21,13 @@
      * @return the current Gradle build being synchronized
      */
     GradleBuild getGradleBuild();
+
+    /**
+     * Registers a problem during initialization. Instead of implementing separate error handling,
+     * project configurators should use this method to report issues.
+     *
+     * @param message the error message describing the problem
+     * @param exception The exception to report; can be {@code null}
+     */
+    void onError(String message, Exception exception);
```

Comment:
Should just be `error`. `onError` would be a name for a method that configures error handling.